frontend/src/lib/components/SelectBox.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useRef, useEffect, useState } from 'react'
+import { kea, useActions, useValues } from 'kea'
+import { Col, Row, Input, Divider } from 'antd'
+import { actionFilterDropdownLogicType } from 'types/scenes/insights/ActionFilter/ActionFilterDropdownType'
+import { List } from 'antd'
+import { DownOutlined, RightOutlined } from '@ant-design/icons'
+import { EntityTypes } from '../../scenes/insights/trendsLogic'
+import Fuse from 'fuse.js'
+import { ActionType } from '~/types'
+
+export interface SelectBoxItem {
+    dataSource: SelectedItem[]
+    renderInfo({ item }: { item: SelectedItem }): JSX.Element
+    name: JSX.Element | string
+}
+
+export interface SelectedItem {
+    name: string
+    key: string
+    action?: ActionType
+    event?: string
+    volume?: number
+    usage_count?: number
+    category?: string
+}
+
+const scrollUpIntoView = (key: string): void => {
+    const searchList = document.querySelector('.search-list')
+    const item = document.querySelector('.action-filter-dropdown [datakey="' + key + '"]')
+    const diff = item?.getBoundingClientRect().top - searchList?.getBoundingClientRect().top
+    if (diff - 30 < 0) searchList.scrollTop = searchList.scrollTop + diff - 30
+}
+const scrollDownIntoView = (key: string): void => {
+    const searchList = document.querySelector('.search-list')
+    const item = document.querySelector('.action-filter-dropdown [datakey="' + key + '"]')
+    const diff = item?.getBoundingClientRect().top - searchList?.getBoundingClientRect().bottom
+    if (diff + 30 > 0) searchList.scrollTop = searchList.scrollTop + diff + 30
+}
+
+export const actionFilterDropdownLogic = kea({
```

Comment:
I think this should move under `lib/logic`?

Reply:
Hm I have a personal preference for having it in the same file, as this is the only component that uses this logic. Having it in the same file means you can see all the logic in the same place.

Reply:
Actually I can't create the logic files without doing it in a separate file so ignore me